segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerEventProcessorImpl.java

```diff
@@ -0,0 +1,308 @@
+ * identified by name. Once this service is started, it will continuously perform processing iterations tailing the
+ * internal Segments of the registered {@link ContainerEventProcessor.EventProcessor}s (every ITERATION_SLEEP ms) and
+ * (safely) invoking their respective handler functions. This class durably stores the point at which the last successful
+ * events has been processed in the form of an attribute in the internal Segment (i.e., LAST_PROCESSED_EVENT_OFFSET).
+ * That is, before reading new events this attribute is read and after a successful processing of events this attribute
+ * is updated with the value of the next position to read. This class also truncates the internal Segments of the registered
+ * {@link ContainerEventProcessor.EventProcessor} based on that attribute.
+ */
+@Slf4j
+public final class ContainerEventProcessorImpl extends AbstractIdleService implements ContainerEventProcessor {
+
+    private static final UUID LAST_PROCESSED_EVENT_OFFSET = new UUID(0, 0);
+    private static final Duration ITERATION_SLEEP = Duration.ofMillis(100); // Execute 10 iterations per second at most.
+    private static final Duration OPERATION_TIMEOUT = Duration.ofMillis(20);
+
+    private final int containerId;
+    private final Map<String, EventProcessorImpl> eventProcessorMap = new ConcurrentHashMap<>();
+    private final Function<String, CompletableFuture<DirectSegmentAccess>> segmentSupplier;
+    private final AtomicBoolean closed;
+    private final ScheduledExecutorService executor;
+
+    //region Constructor
+
+    ContainerEventProcessorImpl(int containerId, @NonNull Function<String, CompletableFuture<DirectSegmentAccess>> segmentSupplier,
+                                @NonNull ScheduledExecutorService executor) {
+        this.containerId = containerId;
+        this.segmentSupplier = segmentSupplier;
+        this.executor = executor;
+        this.closed = new AtomicBoolean(false);
+    }
+
+    //endregion
+
+    //region ContainerEventProcessor Implementation
+
+    @Override
+    public EventProcessor forConsumer(@NonNull String name, @NonNull Function<List<BufferView>, CompletableFuture<Void>> handler,
+                                      @NonNull EventProcessorConfig config) {
+        // If the EventProcessor is already loaded, just return it.
+        if (eventProcessorMap.containsKey(name)) {
+            return eventProcessorMap.get(name);
+        }
+
+        // Instantiate the EventProcessor and put it into the map.
+        DirectSegmentAccess segment = segmentSupplier.apply(name).join();
+        eventProcessorMap.put(name, new EventProcessorImpl(name, segment, handler, config));
+        return eventProcessorMap.get(name);
+    }
+
+    class EventProcessorImpl extends ContainerEventProcessor.EventProcessor {
+
+        // Maximum amount of bytes we allow to be non-processed before start rejecting add requests (4MBs).
+        private final static int MAX_OUTSTANDING_BYTES = 4 * 1024 * 1024;
```

Comment:
config

Reply:
Added this property to `EventProcessorConfig`.